pkg/cidata/user-data.TEMPLATE

```diff
@@ -18,11 +18,141 @@ users:
     {{- end}}
 
 write_files:
+ - content: |
+      #!/bin/sh
+      # This script pretends that /bin/ash can be used as /bin/bash, so all following
+      # cloud-init scripts can use `#!/bin/bash` and `set -o pipefail`.
+      test -f /etc/alpine-release || exit 0
+
+      # Redirect bash to ash (built with CONFIG_ASH_BASH_COMPAT) and hope for the best :)
+      # It does support `set -o pipefail`, but not `[[`.
+      # /bin/bash can't be a symlink because /bin/ash is a symlink to /bin/busybox
+      cat >/bin/bash <<'EOF'
+      #!/bin/sh
+      exec /bin/ash "$@"
+      EOF
+      chmod +x /bin/bash
```

Comment:
Why not just `if [ ! -e /bin/bash ]; then ln -s ash /bin/bash; fi`

Reply:
Also curious, do we really need bash?

Reply:
See the comment in line 29:

```
# /bin/bash can't be a symlink because /bin/ash is a symlink to /bin/busybox
```

busybox does not implement `bash`:

```console
lima-lima:/bin# busybox bash
bash: applet not found
```

So this cannot work:

```console
lima-lima:/bin# rm /bin/bash
lima-lima:/bin# ln -s /bin/ash /bin/bash
lima-lima:/bin# bash -c uname
bash: applet not found
```

Reply:
> Also curious, do we really need bash?

Normal `/bin/sh` does not support `set -o pipefail`, so we couldn't use that in any of the scripts. `/bin/ash` supports it because it is compiled in bash-compatibility mode.

I added the `/bin/bash` wrapper because I didn't want to templatize the `#!/bin/bash` line in all the scripts. That would also require the user to specify the shell in the limayaml because we don't know if the image has `/bin/bash` until it is running, at which time cloud-config has already been processed.

I generally try to avoid adding new config options if reasonably possible; they will eventually become unwieldy even when you practise restraint. 😄